### ISSUE TITLE
How to properly import request.js along with its dependencies in an HTML file for browsers?

### ISSUE THREAD

**Issue by armsp** [octokit/request.js#109]
I am sorry to be asking this here, but I have not been able to run the codes in the README. I sought [help](https://stackoverflow.com/questions/59966543/how-to-use-request-js-by-octokit-on-browser) on SO too but I am afraid nothing came of it.  

I have the following html file (sans the api token) -
```html
<head>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
</head>
<body>
    <script type="module">
        import { request } from "https://cdn.pika.dev/@octokit/request";
    </script>
    <script>
        const octokitRequest = require('@octokit/request');
        myAsyncMethod()
        async function myAsyncMethod () {
            const result = await request("GET /users/:user/repos", {
                headers: {
                    authorization: "token <your token>"
                },
                user: "armsp"
                });
            console.log(`${result.data.length} repos found.`);
        }
    </script>  
</body>
```
This is the error I get -  
```
require.min.js:1 Uncaught Error: Module name "@octokit/request" has not been loaded yet for context: _. Use require([])
https://requirejs.org/docs/errors.html#notloaded
    at makeError (require.min.js:1)
    at Object.s [as require] (require.min.js:1)
    at requirejs (require.min.js:1)
    at github-api-test.html:11

github-api-test.html:1 Uncaught SyntaxError: The requested module '/universal-user-agent/^4.0.0/es2019/universal-user-agent.js' does not provide an export named 'default'
```
I tried to download the js file itself and include it in script tag the usual way, but I don't see any `request.js` file in `src` folder, all are `.ts` files. I don't know how to make sense of it.

Any help would be appreciated.

**Comment by wolfy1339** [comment]
The source code is in TypeScript, and is transpiled at release time for all the various packages, that's why you couldn't find that file in the source.
Can you try removing the  `const octokitRequest` line? That's what seems to be causing that error you shared.

**Comment by gr2m** [comment]
Hmm what should be working is this

```html
<script type="module">
  import { request } from "https://cdn.pika.dev/@octokit/request";

  myAsyncMethod();

  async function myAsyncMethod() {
    const result = await request("GET /users/:user/repos", {
      headers: {
        authorization: "token <your token>"
      },
      user: "armsp"
    });
    console.log(`${result.data.length} repos found.`);
  }
</script>
```

The `import` is async, you have to put the code depending on it into the same `<script type="module">` tag.

However something is not working correctly on Pika's site. `https://cdn.pika.dev/@octokit/request` used to be a pre-bundled single file import, now it has itself a ton of imports and something breaks there. I'll check in with the Pika folks. Sorry for the trouble

**Comment by gr2m** [comment]
I made a demo page at https://github.com/gr2m/octokit-request-browser-test to reproduce the error

**Comment by gr2m** [comment]
The problem has been resolved now. My demo at https://gr2m.github.io/octokit-request-browser-test/ works as expected

**Comment by armsp** [comment]
@gr2m It does indeed work now. May I ask what did you do to make it work? Because I don't really see any major changes in the code. Apologies for asking such simple questions, I am fairly new to web development.

**Comment by gr2m** [comment]
No worries, ask away :)

There was a bad release of `@octokit/request`. In version 5.3.0, the `package.json` file in the published package did not have a "module" key, so Pika couldn't handle it accordingly. You can see the file here:

https://unpkg.com/browse/@octokit/request@5.3.0/package.json

It's fixed in version 5.3.1

https://unpkg.com/browse/@octokit/request@5.3.1/package.json

The folks from Pika and I tried to figure out how that happened, as I'm using their tools to build the package before publishing, but we couldn't figure it out 🤷‍♂

**Comment by armsp** [comment]
@gr2m Got it, Thanks a lot !
